### PR TITLE
Apply liquid template on build if _theme is specified

### PR DESCRIPTION
**Pull request by yufeih**
This PR integrate liquid template into main build pipeline. Static rendering using docs liquid template is still a WIP due to missing custom tags

#3417

### CODE REVIEW

File: src/docfx/build/legacy/metadata/LegacyMetadata.cs [hunk 1]
```diff
@@ -120,46 +118,9 @@ public static JObject GenerateLegacyRedirectionRawMetadata(Docset docset, PageMo
...
-                    if (key.Equals("robots", StringComparison.OrdinalIgnoreCase))
-                    {
-                        // Add to the first line
-                        key = "ROBOTS";
-                        pageMetadataOutput.Insert(0, $"<meta name=\"{key}\" content=\"{HttpUtility.HtmlEncode(content)}\" />{Environment.NewLine}");
-                    }
-                    else
-                    {
-                        pageMetadataOutput.AppendLine($"<meta name=\"{key}\" content=\"{HttpUtility.HtmlEncode(content)}\" />");
-                    }
```
Comment (superyyrrzz) [inline src/docfx/build/legacy/metadata/LegacyMetadata.cs]: Missed this part after move?

Comment (yufeih) [reply src/docfx/build/legacy/metadata/LegacyMetadata.cs]: this is intentional, the order and casing does not matter in html <meta> tags. If it is for comparison, it should be done in patch tool.

Comment (superyyrrzz) [reply src/docfx/build/legacy/metadata/LegacyMetadata.cs]: Another request is to make the "robots" upper case.

Comment (yufeih) [reply src/docfx/build/legacy/metadata/LegacyMetadata.cs]: sitemap requires it to be either `robots` or `ROBOTS`. Treat all metadata keys case insensitive. And make 'robots' as a known metadata for build.

File: src/docfx/template/Template.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        private readonly MustacheTemplate _mustache;
+        private readonly JavaScript _js;
+
+        public Template(string templateDir)
+        {
+            var contentTemplateDir = Path.Combine(templateDir, "ContentTemplate");
+
+            _liquid = new LiquidTemplate(templateDir);
+            _js = new JavaScript(contentTemplateDir);
+            _mustache = new MustacheTemplate(contentTemplateDir);
```
Comment (live1206) [inline src/docfx/template/Template.cs]: >_mustache [](start = 12, length = 9)

never used?

Comment (yufeih) [reply src/docfx/template/Template.cs]: yes, for now

**Pull request merged**